app/uk/gov/hmrc/cataloguefrontend/users/CreateUserController.scala

```diff
@@ -0,0 +1,157 @@
+
+@Singleton
+class CreateUserController @Inject()(
+  override val auth        : FrontendAuthComponents,
+  override val mcc         : MessagesControllerComponents,
+  createUserPage           : CreateUserPage,
+  createUserRequestSentPage: CreateUserRequestSentPage,
+  userManagementConnector  : UserManagementConnector
+) (implicit
+  override val ec: ExecutionContext
+) extends FrontendController(mcc)
+  with CatalogueAuthBuilders
+  with play.api.i18n.I18nSupport {
+
+  private val logger = Logger(getClass)
+
+  private def createUserPermission(teamName: TeamName): Predicate =
+    Predicate.Permission(Resource.from("catalogue-frontend", s"teams/${teamName.asString}"), IAAction("MANAGE"))
+
+  def requestSent(isServiceAccount: Boolean, givenName: String, familyName: String): Action[AnyContent] = Action { implicit request =>
+    Ok(createUserRequestSentPage(isServiceAccount, givenName, familyName))
+  }
+
+  def createUserLanding(isServiceAccount: Boolean): Action[AnyContent] =
+    auth.authenticatedAction(
+      continueUrl = routes.CreateUserController.createUserLanding(isServiceAccount),
+      retrieval   = Retrieval.locations(resourceType = Some(ResourceType("catalogue-frontend")), action = Some(IAAction("MANAGE")))
+    ).apply { implicit request =>
+      Ok(createUserPage(CreateUserForm.form, cleanseUserTeams(request.retrieval), Organisation.values, isServiceAccount))
+    }
+
+  def createUser(isServiceAccount: Boolean): Action[AnyContent] =
+    auth.authenticatedAction(
+      continueUrl = routes.CreateUserController.createUserLanding(isServiceAccount),
+      retrieval   = Retrieval.locations(resourceType = Some(ResourceType("catalogue-frontend")), action = Some(IAAction("MANAGE")))
+    ).async { implicit request =>
+      (
+        for {
+          form <- EitherT.fromEither[Future](CreateUserForm.form.bindFromRequest().fold(
+                    formWithErrors => {
+                      Left(
+                        BadRequest(
+                          createUserPage(
+                            form             = formWithErrors,
+                            teamNames        = cleanseUserTeams(request.retrieval),
+                            organisations    = Organisation.values,
+                            isServiceAccount = isServiceAccount
+                          )
+                        )
+                      )
+                    },
+                    validForm => Right(validForm)
+                  ))
+          _    <- EitherT.liftF(auth.authorised(Some(createUserPermission(form.team))))
+          res  <- EitherT.right[Result](userManagementConnector.createUser(
+                  form.copy(isServiceAccount = isServiceAccount)
+                  ))
+          _    = logger.info(s"user management result: $res:")
+        } yield Redirect(uk.gov.hmrc.cataloguefrontend.users.routes.CreateUserController.requestSent(isServiceAccount, form.givenName, form.familyName))
```

Comment:
You could maybe get away with not using an either here as the fold bindFromRequest behaves in a similar way:

`        CreateUserForm.form.bindFromRequest().fold(
          formWithErrors => {
            Future.successful(
              BadRequest(
                createUserPage(
                  form = formWithErrors,
                  teamNames = cleanseUserTeams(request.retrieval),
                  organisations = Organisation.values,
                  isServiceAccount = isServiceAccount
                )
              )
            )
          },
          createUserRequest => {
            for {
              _   <- auth.authorised(Some(createUserPermission(createUserRequest.team)))
              res <- userManagementConnector.createUser(createUserRequest.copy(isServiceAccount = isServiceAccount))
            } yield {
              logger.info(s"user management result: $res:")
              Redirect(uk.gov.hmrc.cataloguefrontend.users.routes.CreateUserController.requestSent(isServiceAccount, createUserRequest.givenName, createUserRequest.familyName))
            }
          }
        )`